### PR TITLE
Bump eslint and accompanying other libraries

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Upgrade linter and related libraries

Supersedes 
- #1006 
- #979 

🚨 Has failures -- @ministryofjustice/interventions-ui could someone help fix the warnings, please?

```
> hmpps-interventions-ui@0.0.1 lint
> eslint . --cache --max-warnings 0 && stylelint **/*.scss


/Users/davidlantos/Work/ministryofjustice/hmpps-interventions-ui/integration_tests/plugins/index.js
  1:1  error  Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')  import/no-import-module-exports
  2:1  error  Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')  import/no-import-module-exports
  3:1  error  Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')  import/no-import-module-exports
  4:1  error  Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')  import/no-import-module-exports
  5:1  error  Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')  import/no-import-module-exports
  6:1  error  Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')  import/no-import-module-exports

/Users/davidlantos/Work/ministryofjustice/hmpps-interventions-ui/server/routes/appointments/feedback/actionPlanSessions/attendance/actionPlanPostSessionAttendanceFeedbackPresenter.ts
  13:5  error  Default parameters should be last  default-param-last
  14:5  error  Default parameters should be last  default-param-last

/Users/davidlantos/Work/ministryofjustice/hmpps-interventions-ui/server/routes/appointments/feedback/initialAssessment/attendance/initialAssessmentAttendanceFeedbackPresenter.ts
  13:5  error  Default parameters should be last  default-param-last
  14:5  error  Default parameters should be last  default-param-last

/Users/davidlantos/Work/ministryofjustice/hmpps-interventions-ui/server/services/healthCheck.test.ts
  40:7  error  Return values from promise executor functions cannot be read  no-promise-executor-return
  51:7  error  Return values from promise executor functions cannot be read  no-promise-executor-return
```

## What is the intent behind these changes?

To keep up-to-date with latest linter so we have increased confidence in changes going well

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
       "extends": [
         "plugin:@typescript-eslint/eslint-recommended",
         "plugin:@typescript-eslint/recommended",
-        "prettier/@typescript-eslint",
+        "prettier",
```
Comment (sldblog) [inline .eslintrc.json:34]: https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md#version-800-2021-02-21

**Comment by sldblog** [comment]
[Upgrade eslint to version 8](https://trello.com/c/YdreoET3/153-upgrade-eslint-to-version-8)

**Comment by andrewburrowsmoj** [comment]
@sldblog : I left some comments on an older PR that was closed: https://github.com/ministryofjustice/hmpps-interventions-ui/pull/925

I don't know if it's the case anymore but it requires an update of typescript-eslint to be compatible with latest elint. Last time I checked the fix wasn't available yet.

File: package.json [hunk 2]
```diff
@@ -154,14 +154,14 @@
     "@types/redis": "^2.8.32",
     "@types/superagent": "^4.1.13",
     "@types/supertest": "^2.0.11",
-    "@typescript-eslint/eslint-plugin": "^4.33.0",
-    "@typescript-eslint/parser": "^4.33.0",
+    "@typescript-eslint/eslint-plugin": "^5.4.0",
+    "@typescript-eslint/parser": "^5.4.0",
```
Comment (sldblog) [inline package.json:158]: @andrewburrowsmoj afaik that issue is fixed by this bump -- we don't have build errors now, instead, it seems to be genuine linting problems with new rules

**Comment by andrewburrowsmoj** [comment]
@sldblog : fixed the eslint errors.
It's failing on Synk test

**Comment by andrewburrowsmoj** [comment]
@sldblog passing now. I've approved. feel free to merge

**Review by andrewburrowsmoj — APPROVED** [review]

**Comment by sldblog** [comment]
amazing, thank you!

**Pull request merged**